litellm/utils.py

```diff
@@ -726,241 +800,224 @@ def wrapper(*args, **kwargs):  # noqa: PLR0915
         if "litellm_call_id" not in kwargs:
             kwargs["litellm_call_id"] = str(uuid.uuid4())
 
-        model: Optional[str] = None
-        try:
-            model = args[0] if len(args) > 0 else kwargs["model"]
-        except Exception:
-            model = None
-            if (
-                call_type != CallTypes.image_generation.value
-                and call_type != CallTypes.text_completion.value
-            ):
-                raise ValueError("model param not passed in.")
+        model = _get_model_from_wrapper_args(
+            call_type=original_function.__name__,
+            empty_model_call_types=[
+                CallTypes.image_generation.value,
+                CallTypes.text_completion.value,
+            ],
+            args=args,
+            kwargs=kwargs,
+        )
 
-        try:
```

Comment:
**The majority of the diff in wrapper below this line is just un-indenting the try / catch block now that retry logic from Router is used instead**